src/printer.js

```diff
@@ -2934,6 +3035,26 @@ function isEmptyBlock(doc) {
   return str === "{}";
 }
 
+
+function shouldTypeScriptTypeAvoidColon(path) {
+  // As the special TS nodes isn't returned by the node helpers,
+  // we use the stack directly to get the parent node.
+  const parent = path.stack[path.stack.length - 3]
```

Comment:
This is the only thing that looks magical in this PR. I'd love if we can get a way to avoid this hack. In the meantime, let's pull it in since it is working.